src/monodroid/jni/pinvoke-override-api.cc

```diff
@@ -0,0 +1,709 @@
+			log_info_nocheck (category, message);
+			break;
+	}
+}
+
+static void
+monodroid_free (void *ptr)
+{
+        free (ptr);
+}
+
+static int
+_monodroid_max_gref_get ()
+{
+        return static_cast<int>(androidSystem.get_max_gref_count ());
+}
+
+static int
+_monodroid_gref_get ()
+{
+	return osBridge.get_gc_gref_count ();
+}
+
+
+static void
+_monodroid_gref_log (const char *message)
+{
+        osBridge._monodroid_gref_log (message);
+}
+
+static int
+_monodroid_gref_log_new (jobject curHandle, char curType, jobject newHandle, char newType, const char *threadName, int threadId, const char *from, int from_writable)
+{
+        return osBridge._monodroid_gref_log_new (curHandle, curType, newHandle, newType, threadName, threadId, from, from_writable);
+}
+
+static void
+_monodroid_gref_log_delete (jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+        osBridge._monodroid_gref_log_delete (handle, type, threadName, threadId, from, from_writable);
+}
+
+static void
+_monodroid_weak_gref_new (jobject curHandle, char curType, jobject newHandle, char newType, const char *threadName, int threadId, const char *from, int from_writable)
+{
+        osBridge._monodroid_weak_gref_new (curHandle, curType, newHandle, newType, threadName, threadId, from, from_writable);
+}
+
+static void
+_monodroid_weak_gref_delete (jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+        osBridge._monodroid_weak_gref_delete (handle, type, threadName, threadId, from, from_writable);
+}
+
+static void
+_monodroid_lref_log_new (int lrefc, jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+        osBridge._monodroid_lref_log_new (lrefc, handle, type, threadName, threadId, from, from_writable);
+}
+
+static void
+_monodroid_lref_log_delete (int lrefc, jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+        osBridge._monodroid_lref_log_delete (lrefc, handle, type, threadName, threadId, from, from_writable);
+}
+
+static void
+_monodroid_gc_wait_for_bridge_processing ()
+{
+        mono_gc_wait_for_bridge_processing ();
+}
+
+static int
+_monodroid_get_android_api_level ()
+{
+        return monodroidRuntime.get_android_api_level ();
+}
+
+static void
+monodroid_clear_gdb_wait ()
+{
+        monodroidRuntime.set_monodroid_gdb_wait (false);
+}
+
+static void*
+_monodroid_get_identity_hash_code (JNIEnv *env, void *v)
+{
+        intptr_t rv = env->CallStaticIntMethod (monodroidRuntime.get_java_class_System (), monodroidRuntime.get_java_class_method_System_identityHashCode (), v);
+        return (void*) rv;
+}
+
+static void*
+_monodroid_timezone_get_default_id ()
+{
+	JNIEnv *env          = osBridge.ensure_jnienv ();
+	jmethodID getDefault = env->GetStaticMethodID (monodroidRuntime.get_java_class_TimeZone (), "getDefault", "()Ljava/util/TimeZone;");
```

Comment:
We want to look these `jmethodID`s up every time, instead of caching them?

Reply:
Yes, we did that to speed up startup. Caching it at the later stage would probably require locks. We can implement that, but not in this PR.